test/test_credential_process_json.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+import json
+from aws_adfs import login
+from mock import patch
+
+
+class TestCredentialProcesJson:
+
+    def setup_method(self, method):
+        self.access_key = 'AKIAIOSFODNN7EXAMPLE'
+        self.secret_key = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYzEXAMPLEKEY'
+        self.session_token = 'AQoDYXdzEPT//////////wEXAMPLEtc764bNrC9SAPBSM22wDOk4x4HIZ8j4FZTwdQWLWsKWHGBuFqwAeMicRXmxfpSPfIeoIYRqTflfKD8YUuwthAx7mSEI/qkPpKPi/kMcGdQrmGdeehM4IC1NtBmUpp2wUE8phUZampKsburEDy0KPkyQDYwT7WZ0wq5VSXDvp75YU9HFvlRd8Tx6q6fE8YQcHNVXAkiY9q6d+xo0rKwT38xVqr7ZD0u0iPPkUL64lIZbqBAz+scqKmlzm8FDrypNC9Yjc8fPOLn9FX9KSYvKTr4rvx3iSIlTJabIQwj2ICCR/oLxBA=='
+        self.expiration = '2020-06-30T20:17:02.439725+00:00'
```

Comment:
One thing that confuses me is that the actual format of the expiration date returned by the AWS session response does not respect the [ISO8601 documentation](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations) linked from [AWS docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html):

> # Combined date and time representations
> A single point in time can be represented by concatenating a complete date expression, the letter _"T"_ as a delimiter, and a valid time expression. For example, "2007-04-05T14:30". It is permitted to omit the _"T"_ character by mutual agreement as in "200704051430". **Separating date and time parts with other characters such as space is _not_ allowed in ISO 8601, but allowed in its profile RFC 3339**.

Reply:
@pdecat - I noticed that as well. It looks like they're not really as strict as they state in the docs. 

Regardless, the latest changeset outputs strict ISO8601